app/controllers/api_controller/providers.rb

```diff
@@ -5,9 +5,14 @@ module Providers
     CREDENTIALS_ATTR  = "credentials"
     AUTH_TYPE_ATTR    = "auth_type"
     DEFAULT_AUTH_TYPE = "default"
-    ENDPOINT_ATTRS    = %w(hostname ipaddress port)
+    ENDPOINT_ATTRS    = %w(hostname ipaddress port connection_configurations).freeze
     RESTRICTED_ATTRS  = [TYPE_ATTR, CREDENTIALS_ATTR, ZONE_ATTR, "zone_id"]
 
+    def show_providers
```

Comment:
Its called by using method_missing on the api_controller, if its not set then it will use the one in generic.rb and give a basic show, if you want to expand the basic object you need to override it and add fields.